src/base/static/components/form-field/field-definitions.js

```diff
@@ -56,6 +68,17 @@ const getSharedFieldProps = (fieldConfig, context) => {
   };
 };
 
+const getCheckboxLabels = (fieldValue, fieldConfig) => {
+  if (!ImmutableList.isList(fieldValue)) {
```

Comment:
I see where you're coming from here. I'm not sure-- I guess the advantage of putting this logic in `field-definitions.js` is that it lives in the same place where we already break out each field type. If we move that logic to where we serialize the model we'd have to include a special check for the checkbox field type. So the current serialization:
```
fromJS(this.props.model.attributes)
```
would become something like:
```
fromJS(this.props.model.attributes).map((fieldValue, fieldName) => {
    const fieldType = this.props.fieldConfig.find(field => field.name === fieldName).type;
    if (
        fieldType === constants.BIG_CHECKBOX_FIELD_TYPENAME &&
        !ImmutableList.isList(fieldValue)
    ) {
        fieldValue = ImmutableList(fieldValue);
    }

    return fieldValue;
})
```
(That might not be exactly right, but something along those lines.)

Which do you think would be preferable?

Reply:
>If we move that logic to where we serialize the model we'd have to include a special check for the checkbox field type.

I don't think that's necessary. In PlaceFormView, we could be performing a "deep serialization" of the model attributes - whether it's a JS hash to an ImmutableMap or an array to an ImmutableList. So instead of having to checking the field type, we check whether the type is an array:


```
    if (Array.isArray(fieldValue)) {
        fieldValue = ImmutableList(fieldValue);
    }
```

Does that make sense?

Reply:
Yes-- in fact we are doing a deep serialization (that's the default behavior of `fromJS()`), but the issue is that responses to checkboxes might be either a string or an array, depending on whether a single option or multiple options were checked. We convert a string to an array of length one so it works as expected in `big-checkbox-field-response.js`.

But now that I think about it, maybe `big-checkbox-field-response.js` should just handle that discrepancy itself?

Reply:
So the only point of this conditional is to catch cases where a single option was checked, in which case the value is a string? It seems like that is a result of faulty logic with the way we are storing the checkbox responses - it should always be an array, even if only one item was selected. 

Fixing that issue is outside the scope of this PR, but we should definitely add a comment to explain why `fieldValue` is a string.

Also, after revisiting this, I think we can simplify the way we are fetching our Response field components. I think we can have this field-definitions object only return the component to our FormFieldResponse, ie: `getResponseComponent: () => BigRadioFieldResponse`. Then, in FormFieldResponse, we have something like this:

```
const FieldResponseComponent = fieldDefinitions[props.fieldConfig.type].getResponseComponent()
return (
  <div>
    {FieldResponseComponent && 
      <FieldResponseComponent 
          label={fieldConfig.content.find(option => option.value === props.fieldValue).label}
          value={props.fieldValue}
          backboneAttachmentModelsAttributes={props.backboneAttachmentModelsAttributes}
          labels={getCheckboxLabels(props.fieldValue, fieldConfig)}
       />
    }
  </div>
)
```

That way, FieldResponseComponent will be the owner of normalizing the data - not field-definition nor the individual field response type components. Also, it keeps field-definitions simple and helps prevent it from becoming a dumping grounds of miscellaneous logic :-)

Does that make sense?

Reply:
> It seems like that is a result of faulty logic with the way we are storing the checkbox responses - it should always be an array, even if only one item was selected.

Agreed. Unfortunately we have to support both formats to make sure existing checkbox data renders correctly. But we could refactor the checkbox input type to iron out this discrepancy on any new checkbox data.

Re simplifying `field-definitions.js`: yes, that makes sense to me. I guess I would lean toward just passing along a common set of props to each response component, since the props footprint is pretty minimal for the response components anyway.